### PR TITLE
Improve RubyLex's tests

### PR DESCRIPTION
**Pull request by st0012**
When doing code reviews about RubyLex, it often took me a while to understand the test helpers' usage, and then the test's intention. So I think it may be worth refactoring the helpers a bit to make it easier for everyone.

It'd be easier to review/test the changes by commits.

(I know there are further refactors we can do, but I want to see how you think about it first)

### CODE REVIEW

**Comment by hasumikin** [comment]
@st0012 
My poor knowledge of RubyLex can't help me to judge whether this refactoring is effective.
If you want to make things fast, I'm OK that you go ahead.
If you don't hurry, give me more several weeks to catch up :)

**Comment by st0012** [comment]
It's not urgent by any means. My intention is just to help whoever works on `RubyLex` feels easier to write its tests.

**Comment by st0012** [comment]
@tompng Given you're most familiar with RubyLex and its tests, could you give this change a look and let me know if it'll help your work?

**Review by tompng — CHANGES_REQUESTED** [review]
The change looks good!
We always test both current_line_spaces and new_line_spaces, so the test seems clearer than before :+1:

File: test/irb/test_ruby_lex.rb [hunk 1]
```diff
@@ -29,22 +27,60 @@ def teardown
...
       context.auto_indent_mode = true
+
       ruby_lex = RubyLex.new(context)
       io = MockIO_AutoIndent.new([lines, last_line_index, byte_pointer, add_new_line]) do |auto_indent|
         error_message = "Calculated the wrong number of spaces for:\n #{lines.join("\n")}"
         assert_equal(correct_space_count, auto_indent, error_message)
       end
+
       ruby_lex.set_input(io)
       ruby_lex.set_auto_indent
```
[the remark this answers is not in the document]
Comment (st0012) [reply test/irb/test_ruby_lex.rb:48]: Good call 👍 Updated

File: test/irb/test_ruby_lex.rb [hunk 2]
```diff
@@ -29,20 +30,59 @@ def teardown
...
       lines = lines + [""] if add_new_line
       last_line_index = lines.length - 1
       byte_pointer = lines.last.length
 
       context = build_context
       context.auto_indent_mode = true
+
       ruby_lex = RubyLex.new(context)
-      io = MockIO_AutoIndent.new([lines, last_line_index, byte_pointer, add_new_line]) do |auto_indent|
+      mock_io = MockIO_AutoIndent.new(lines, last_line_index, byte_pointer, add_new_line) do |auto_indent|
```
Comment (tompng) [inline test/irb/test_ruby_lex.rb:42]: The block passed to `MockIO_AutoIndent.new` is not used because `&assertion` is deleted from parameter

Comment (st0012) [reply test/irb/test_ruby_lex.rb:42]: 🤦‍♂️ Because I've moved the assertion out to `assert_row_indenting` and this is not needed.

**Review by tompng — APPROVED** [review]

**Pull request merged**